sb3_contrib/common/utils.py

```diff
@@ -114,16 +114,12 @@ def conjugate_gradient_solver(
 
     p = residual.clone()
 
-    for i in range(max_iter):
-        # A @ p (matrix vector multiplication)
-        A_dot_p = matrix_vector_dot_fn(p)
-
-        alpha = residual_squared_norm / p.dot(A_dot_p)
-        x += alpha * p
-
-        if i == max_iter - 1:
```

Comment:
mypy didn't like early return?

I need to check that one, but I think it is ok sometimes to ignore mypy warnings/errors if it is for readability.

Reply:
mypy think the function could never return

Reply:
is there a case where it would never return?
if `max_iter=0`, the loop doesn't run, if `max_iter=1`, then it returns automatically, I guess it is a false positive?

Reply:
>  it is ok sometimes to ignore mypy warnings/errors if it is for readability.

Agree. But here, I've find the new version more readable.

Reply:
oh, I see, maybe `assert max_iter > 0` would be a better fix?

Reply:
> if max_iter=0, the loop doesn't run

In that cas it is supposed to return None then. Is it the desired behaviour?

Reply:
```python
    assert max_iter > 0
    for i in range(max_iter):
        if i == max_iter - 1:
            return x
```

mypy:
```shell
error: Missing return statement  [return]
```

Reply:
mmh, probably too advanced for mypy, I guess we can ignore it then

Reply:
> In that case it would return None then. Is it the desired behaviour?

calling the CG solver with `max_iter=0` doesn't make much sense, but adding `return x` at the end should solve the issue, no? (so not changing the rest, even though the return statement will almost never be used)

Reply:
> adding `return x` at the end should solve the issue, no?

It would.

The following two structure would work.

1.
```python
for i in range(max_iter):
    # Do A
    if i == max_iter - 1:
         return 
    # Do B
return 
```

2.
```python
# Do A
for _ in range(max_iter - 1):
    # Do B
    # Do A
return
```

I find the second one more readable, but if you prefer to limit the number of changes, we can keep the first one.